### PR TITLE
[Merged by Bors] - feat: infimum of filter coproduct and filter product

### PR DESCRIPTION
**Pull request by trivial1711**
We prove that if `f₁, f₂` are filters on a type α and `g₁, g₂` are filters on a type β, then `f₁.coprod g₁ ⊓ f₂ ×ˢ g₂ ≤ f₁ ×ˢ g₂ ⊔ f₂ ×ˢ g₁`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
Seems reasonable 

maintainer merge

File: Mathlib/Order/Filter/Prod.lean [hunk 1]
```diff
@@ -455,6 +455,15 @@ theorem coprod_neBot_left [NeBot f] [Nonempty β] : (f.coprod g).NeBot :=
 theorem coprod_neBot_right [NeBot g] [Nonempty α] : (f.coprod g).NeBot :=
   coprod_neBot_iff.2 (Or.inr ⟨‹_›, ‹_›⟩)
 
+theorem coprod_inf_prod_le (f₁ f₂ : Filter α) (g₁ g₂ : Filter β) :
+    f₁.coprod g₁ ⊓ f₂ ×ˢ g₂ ≤ f₁ ×ˢ g₂ ⊔ f₂ ×ˢ g₁ := calc
+  f₁.coprod g₁ ⊓ f₂ ×ˢ g₂
+  _ = (f₁ ×ˢ ⊤ ⊔ ⊤ ×ˢ g₁) ⊓ f₂ ×ˢ g₂            := by rw [coprod_eq_prod_top_sup_top_prod]
+  _ = f₁ ×ˢ ⊤ ⊓ f₂ ×ˢ g₂ ⊔ ⊤ ×ˢ g₁ ⊓ f₂ ×ˢ g₂   := inf_sup_right _ _ _
+  _ = (f₁ ⊓ f₂) ×ˢ g₂ ⊔ f₂ ×ˢ (g₁ ⊓ g₂)         := by simp [prod_inf_prod]
+  _ ≤ f₁ ×ˢ g₂ ⊔ f₂ ×ˢ g₁                        :=
```
Comment (ADedecker) [inline Mathlib/Order/Filter/Prod.lean:464]:
```suggestion
  _ ≤ f₁ ×ˢ g₂ ⊔ f₂ ×ˢ g₁                       :=
```

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**